### ISSUE TITLE
Validate YAML format when using the -v option

### ISSUE THREAD

**Issue by jvanzyl** [oneops/boo#35]
To help users make sure their configurations are valid, perform validation on the YAML with the -v option.

**Comment by nyang64** [comment]
yes.  that -v option is very useful in identifying the yaml file format issue, such as the indentation of configuration sections.

**Comment by sureshg** [comment]
@nyang64 Doesn't the format and indentation errors are handled by the YAML parser library (`jackson-dataformat-yaml`) by default?. ie, It throws an error when loading an invalid YAML. I think we need more validation for the configuration used in YAML based on the component metadata. We were debugging one issue yesterday, where user configured the `postgresql` port as an `integer` value, but the metadata/component was expecting a`String`. Errors like these can't be validated by yaml parser (because it's  a valid yaml) without understanding the component/pack metadata.

**Comment by jvanzyl** [comment]
Up to the point where the YAML is viewed it has only incorporated the ~/.boo/config and been interpolated and not yet undergone deserialization by Jackson. So we can certainly incorporate that earlier for validation of the YAML format. Validation of the specification of packs, components and their configuration could be an additional round of validation but it's only possible to get this information hitting the CMS API directly which we probably don't want.
